docs/_data/button-groups.json

```diff
@@ -6,9 +6,9 @@
       "description": "Base button group style."
     },
     {
-      "class": ".s-btn-group--container",
```

Comment:
So, I originally intended this to be a non-breaking change, but a deprecation. In 77c425c9c1f68cb8d06015f89a58ba180b22ee6d, I've targeted the `form` element and once again made this a deprecation. Basically with the PR as it is now, the changes should be backwards compatible. 



![image](https://user-images.githubusercontent.com/647177/201414111-5a47a9f5-12d5-4859-89ed-4c39eb873f5c.png)

> **Note**
> Image for demo purposes only. Don't worry. I didn't change the docs to this permanently.

Should we reintroduce `.s-btn-group--container`? I know targeting a HTML tag feels a little icky but it doesn't feel totally wrong here. That said, I'm fine with targeting that class and punting on any of this and FWIW it's pretty low stakes since I only found it used in [one place in Core](https://github.com/StackEng/StackOverflow/blob/master/StackOverflow/Views/Channels/Admin/UserManagement/_UserSyncSettings.cshtml#L79-L88).

---

PS: `.s-btn-group--container` previously just set any nested button border radius to 0. That means if the button was at the start or end of the button group, it would get squared off. I've addressed this in 77c425c9c1f68cb8d06015f89a58ba180b22ee6d.